test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +129,52 @@ def test_t(self):
         self.assertEqual(tranposed_mat2.size(), torch.Size([12, 8]))
         self.assertEqual(tranposed_mat2.placements, shard_spec)
 
+    # TODO: Need to investigate why test failed in CPU for baddbmm.
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_sharded_baddbmm(self):
+        # If beta is 0, input tensor will be ignored
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        torch.manual_seed(self.rank)
+        tensor = torch.rand(3, 5, 6, device=self.device_type)
+        batch_1 = torch.rand(3, 5, 8, device=self.device_type)
+        batch_2 = torch.rand(3, 8, 6, device=self.device_type)
+        local_result = torch.baddbmm(
+            tensor, batch_1, batch_2, beta=0.0, alpha=0.5
+        )
+        sharding = [Shard(0)]
```

Comment:
no problem. working on add something similar to https://github.com/pytorch/tau/blob/c4e9d3fc57bd2a624be68da13394fe4b6a9559be/test/spmd/tensor/test_matrix_ops.py#L83 in bmm/baddbmm tests.

Reply:
> Since this bmm/baddmm rule is requiring inputs that is not only sharded on the batched dimension, we need to make it more general here.
> 
> Also suggested by @aazzolini, we should use einsum propagation here instead of manually writing rules to avoid all the special cases. Specifically, we should register the `bmm` by calling into this rule https://github.com/pytorch/PiPPy/blob/main/spmd/tensor/ops/math_ops.py#L21, and `baddmm` rule should also call into this + a pointwise rule.

For 2 tensors of shape "bmk" and "bkn", sharding on dim other than the batched dimension now gives "Two different input dims are sharded across the same mesh dim!" exception which I believe is justified because such a behavior (i.e. sharding on other dimensions) does not make sense. Do i understand correctly?

Reply:
Fixing tests.